core/src/main/scala-3/sttp/tapir/macros/union_derivation.scala

```diff
@@ -0,0 +1,58 @@
+package sttp.tapir.macros
+
+import scala.compiletime.*
+import scala.deriving.*
+import scala.quoted.*
+
+@scala.annotation.implicitNotFound("${A} is not a union of ${T}")
+sealed trait IsUnionOf[T, A]
```

Comment:
can we make all of these `private[tapir]`? the less we expose to the outside world, the better

Reply:
sure, but they are generic and could be useful 🙃

Reply:
Well it's not the focus of the tapir library. The less code to maintain in `core`, the better :) Nobody is goign to include `tapir-core` in their project to get `IsUnionOf`